src/main/java/org/jabref/gui/externalfiles/DuplicateDecisionResult.java

```diff
@@ -0,0 +1,23 @@
+package org.jabref.gui.externalfiles;
+
+import org.jabref.gui.duplicationFinder.DuplicateResolverDialog;
+import org.jabref.model.entry.BibEntry;
+
+public class DuplicateDecisionResult {
```

Comment:
This should be a `record`

Reply:
In IntelliJ, hover on "DuplicateDecisionResult", press <kbd>Alt</kbd>+<kbd>Enter</kbd>, and select "Convert to a record"

![image](https://github.com/JabRef/jabref/assets/1366654/44a2d079-908f-43f2-b2d0-7725927fc3be)

Press <kbd>Enter</kbd>

Commit the changes:

```diff
--- a/src/main/java/org/jabref/gui/externalfiles/DuplicateDecisionResult.java                             
+++ b/src/main/java/org/jabref/gui/externalfiles/DuplicateDecisionResult.java                             
@@ -3,21 +3,6 @@ package org.jabref.gui.externalfiles;                                                    
 import org.jabref.gui.duplicationFinder.DuplicateResolverDialog;                                         
 import org.jabref.model.entry.BibEntry;                                                                  
                                                                                                          
-public class DuplicateDecisionResult {                                                                   
-    private final DuplicateResolverDialog.DuplicateResolverResult decision;                              
-    private final BibEntry mergedEntry;                                                                  
-                                                                                                         
-    public DuplicateDecisionResult(DuplicateResolverDialog.DuplicateResolverResult decision, BibEntry mer
gedEntry) {                                                                                               
-        this.decision = decision;                                                                        
-        this.mergedEntry = mergedEntry;                                                                  
-    }                                                                                                    
-                                                                                                         
-    public DuplicateResolverDialog.DuplicateResolverResult getDecision() {                               
-        return decision;                                                                                 
-    }                                                                                                    
-                                                                                                         
-    public BibEntry getMergedEntry() {                                                                   
-        return mergedEntry;                                                                              
-    }                                                                                                    
+public record DuplicateDecisionResult(DuplicateResolverDialog.DuplicateResolverResult decision, BibEntry 
mergedEntry) {                                                                                            
 }                                                                                                        
                                                                                                          
diff --git a/src/main/java/org/jabref/gui/externalfiles/ImportHandler.java b/src/main/java/org/jabref/gui/
externalfiles/ImportHandler.java                                                                          
index a10277c2c9..64780f5848 100644                                                                       
--- a/src/main/java/org/jabref/gui/externalfiles/ImportHandler.java                                       
+++ b/src/main/java/org/jabref/gui/externalfiles/ImportHandler.java                                       
@@ -227,7 +227,7 @@ public class ImportHandler {                                                          
                                                                                                          
     public BibEntry handleDuplicates(BibDatabaseContext bibDatabaseContext, BibEntry originalEntry, BibEn
try duplicateEntry) {                                                                                     
         DuplicateDecisionResult decisionResult = getDuplicateDecision(originalEntry, duplicateEntry, bibD
atabaseContext);                                                                                          
-        switch (decisionResult.getDecision()) {                                                          
+        switch (decisionResult.decision()) {                                                             
             case KEEP_RIGHT:                                                                             
                 bibDatabaseContext.getDatabase().removeEntry(duplicateEntry);                            
                 break;                                                                                   
@@ -235,7 +235,7 @@ public class ImportHandler {                                                          
                 break;                                                                                   
             case KEEP_MERGE:                                                                             
                 bibDatabaseContext.getDatabase().removeEntry(duplicateEntry);                            
-                return decisionResult.getMergedEntry();                                                  
+                return decisionResult.mergedEntry();                                                     
             case KEEP_LEFT:                                                                              
             case AUTOREMOVE_EXACT:                                                                       
             case BREAK:                                                                                  
```

Reply:
This is the only missing thing to get this in. Can you try to do it?

Reply:
I just did that. Thank you for your hint!